dGame/dUtilities/SlashCommandHandler.cpp

```diff
@@ -810,7 +810,70 @@ void SlashCommandHandler::HandleChatCommand(const std::u16string& command, Entit
         }
     }
 
-	if (chatCommand == "mailitem" && entity->GetGMLevel() >= GAME_MASTER_LEVEL_MODERATOR && args.size() >= 2) {
+	if (chatCommand == "setmailsender" && entity->GetGMLevel() >= GAME_MASTER_LEVEL_DEVELOPER) {
+		// Display help for the user
+		if (args.size() == 0) {
+			ChatPackets::SendSystemMessage(sysAddr, u"Sets a custom sender name of the next mail command used, with a maximum of 20 characters");
+			ChatPackets::SendSystemMessage(sysAddr, u"/setmailsender <custom sender name>");
+			return;
+		}
+
+		std::stringstream ss;
+		for (auto string : args)
+			ss << string << " ";
+		
+		if (ss.str().length() > 20) {
+			ChatPackets::SendSystemMessage(sysAddr, u"Custom sender length is too long. A custom sender name must be 20 characters or less.");
+			return;
+		}
+
+		entity->GetCharacter()->SetMailSender(ss.str());
+		ChatPackets::SendSystemMessage(sysAddr, u"Custom mail sender set.");
+		return;
+	}
+
+	if (chatCommand == "setmailsub" && entity->GetGMLevel() >= GAME_MASTER_LEVEL_DEVELOPER) {
```

Comment:
I'd prefer the use of an `else if ` chain here

Reply:
Hey Mick!

Thanks for the feedback. Just to make sure, you would like me to do an else if chain for the `setmail____` commands? I only ask because this would be inconsistent with what has been used for other commands in the slashcommandhandler (such as `/setanntitle` and `/setannmsg`, which came by default with the original DLU release. I'm willing to do it, I just wanted to bring up that this particular file usually just uses if statements without and if and else if chains.

Reply:
Ah right. Yeah using if-else chains is just neater as we'd never expect the chat command variable to change during execution, leading us to having no use for multiple if statements. You don't have to change the old ones, but it's better to do this going forward imo.

Reply:
Hi Mick,

I have updated all the mail commands to be in an else if chain.